zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -1378,12 +1360,6 @@ impl<'a, F: Field> ConstraintBuilder<'a, F> {
         ret
     }
 
-    pub(crate) fn add_constraints(&mut self, constraints: Vec<(&'static str, Expression<F>)>) {
-        for (name, constraint) in constraints {
-            self.add_constraint(name, constraint);
-        }
-    }
-
     pub(crate) fn add_constraint(&mut self, name: &'static str, constraint: Expression<F>) {
```

Comment:
I think this one is redundant, because it's already implemented via the `ConstrainBuilderCommon` trait impl.

Reply:
add_constraint in singular is not implemented in ConstrainBuilderCommon, I think I didn't do it because both implementations are slightly different

Reply:
ahh, yes, I see it is implemented in the implementation of the trait, yes will fix it